File: API/models/Post.js
const mongoose = require("mongoose");
const Schema = mongoose.Schema;

const Post = new Schema(
  {
    title: { type: String },
    description: { type: String },
    location: {
      type: { type: String },
      coordinates: [Number],
    },
    type: { type: Boolean },
    author: { type: String },
    images: { type: [String] },
    added_on: { type: Date },
  },
  {
    collection: "posts",
  }
);

module.exports = mongoose.model("Post", Post);

File: app/src/main/java/com/example/tracepoint/models/Post.kt
package com.example.tracepoint.models

data class Post(
    val _id: String,
    val title: String,
    val description: String,
    val location: Location,
    val type: Boolean, // true for found, false for lost
    val author: String,
    val images: List<String>
)

data class Location(
    val type: String,
    val coordinates: List<Double>
)

data class CreatePostRequest(
    val title: String,
    val description: String,
    val location: Location,
    val type: Boolean,
    val author: String?,
    val images: List<String>
)
